docs/core/extensions/file-globbing.md

```diff
@@ -0,0 +1,248 @@
+- Calls `Execute` given the `DirectoryInfoWrapper` instance to yield a <xref:Microsoft.Extensions.FileSystemGlobbing.PatternMatchingResult> object.
+
+> [!NOTE]
+> The `DirectoryInfoWrapper` type is defined in the `Microsoft.Extensions.FileSystemGlobbing.Abstractions` namespace, and the `DirectoryInfo` type is defined in the `System.IO` namespace. To avoid unnecessary `using` statements, you can use the provided extension methods.
+
+There is another convenience based extension method that yields an `IEnumerable<string>` representing the matching files:
+
+```csharp
+Matcher matcher = new();
+matcher.AddIncludePatterns(new[] { "*.txt", "*.asciidoc", "*.md" });
+
+string searchDirectory = "../starting-folder/";
+
+IEnumerable<string> matchingFiles = matcher.GetResultsInFullPath(searchDirectory);
+
+// Use matchingFiles if there are any found.
+// The files in this collection are fully qualified file system paths.
+```
+
+The preceding C# code:
+
+- Instantiates a <xref:Microsoft.Extensions.FileSystemGlobbing.Matcher> object.
+- Calls <xref:Microsoft.Extensions.FileSystemGlobbing.MatcherExtensions.AddIncludePatterns(Microsoft.Extensions.FileSystemGlobbing.Matcher,System.Collections.Generic.IEnumerable{System.String}[])> to add several file name patterns to include.
+- Declares and assigns the search directory value.
+- Calls `GetResultsInFullPath` given the `searchDirectory` value to yield all matching files as a `IEnumerable<string>`.
+
+### Match overloads
+
+The <xref:Microsoft.Extensions.FileSystemGlobbing.PatternMatchingResult> object represents a collection of <xref:Microsoft.Extensions.FileSystemGlobbing.FilePatternMatch> instances, and exposes a `boolean` value indicating whether the result has matches &mdash; <xref:Microsoft.Extensions.FileSystemGlobbing.PatternMatchingResult.HasMatches?displayProperty=nameWithType>.
+
+With a `Matcher` instance, you can call the any of the various `Match` overloads to get a pattern matching result. The `Match` methods invert the responsibility on the caller to provide a file or a collection of files in which to evaluate for matches. In other words, the caller is responsible for passing the file to match on.
+
+> [!IMPORTANT]
+> When using any of the `Match` overloads, there is no file system I/O involved. All of the file globbing is done in memory with the include and exclude patterns of the `matcher` instance. The parameters of the `Match` overloads do not have to be fully qualified paths. The current directory (<xref:System.IO.Directory.GetCurrentDirectory?displayProperty=nameWithType>) is used when not specified.
+
+To match a single file:
+
+```csharp
+Matcher matcher = new();
+matcher.AddInclude("**/*.md");
+
+PatternMatchingResult result = matcher.Match("file.md");
+```
+
+The preceding C# code:
+
+- Matches any file with the *.md* file extension, at an arbitrary directory depth.
+- If a file named *file.md* exists in a subdirectory from the current directory:
```

Comment:
The preceding alert box says there's no file system I/O, but this statement seems to contradict that, as it implies that the file system is queried to find out if file.md exists.

Reply:
The match overloads do not involve I/O, they query the in memory directory hierarchy.